packages/block-library/src/image/view.js

```diff
@@ -17,6 +17,43 @@ const focusableSelectors = [
 	'[tabindex]:not([tabindex^="-"])',
 ];
 
+// I've defined the scroll handling outside of the store
+// to prevent confusion as to how this logic is called.
+// This logic is not referenced directly by the directives
+// in the markup because the scroll event we need to listen
+// to is triggered on the window; so we define it outside of
+// the store to signal that the behavior here is different.
+// If we find a compelling reason to move it to the store,
+// feel free to do so.
+let scrollCallback;
+
+// We need to track whether a user is touching the screen so we
+// can handle the scroll override differently on mobile devices.
+let isTouching = false;
+let lastTouchTime = 0;
+
+// Using overflow: hidden to prevent scrolling while the lightbox
+// is open causes the content to shift and prevents the zoom
+// animation from working in some cases because we're unable to
+// account for the layout shift when doing the animation calculations.
+// Instead, here we use JavaScript to prevent and reset the scrolling
+// behavior. In the future, we may be able to use CSS or overflow: hidden
+// instead to not rely on JavaScript, but this seems to be the best approach
+// for now that provides the best visual experience.
+function handleScroll( context ) {
+	// We can't override the scroll behavior on mobile devices
+	// because doing so breaks the pinch to zoom functionality, and we
+	// want to allow users to zoom in further on the high-res image.
+	if ( ! isTouching && Date.now() - lastTouchTime > 450 ) {
+		// We are unable to use event.preventDefault() to prevent scrolling
+		// because the scroll event can't be canceled, so we reset the position instead.
+		window.scrollTo(
+			context.core.image.lastScrollLeft,
+			context.core.image.lastScrollTop
```

Comment:
I guess these are already defined and used, but they feel more like `context.core.image.initialScrollLeft` and `initialScrollTop` to me since they aren't updated; since they are resetting the scroll to the position it was in when initially opening the lightbox.

Reply:
Got it! While `initialScrollTop` and `initialScrollLeft` may be better, they still don't ring true to me, so I've renamed them to `scrollTopReset` and `scrollLeftReset`, which I think more clearly expresses the intent.